cmake/CompilerFlagsHelpers.cmake

```diff
@@ -102,5 +102,8 @@ foreach(COMPILER_LANGUAGE ${SUPPORTED_COMPILER_LANGUAGE_LIST})
     if(CMAKE_${COMPILER_LANGUAGE}_COMPILER_ID STREQUAL "PGI")
       set(CMAKE_${COMPILER_LANGUAGE}_WARNING_ALL "")
     endif()
+    if(CMAKE_${COMPILER_LANGUAGE}_COMPILER_ID MATCHES "Clang|AppleClang")
+      set(CMAKE_${COMPILER_LANGUAGE}_IGNORE_WARNINGS "-Wno-deprecated-register -Wno-writable-strings")
+    endif()
```

Comment:
@alexsavulescu : (you might have already)we should ditch register right after this PR!

Reply:
Yes, on the list. Fairly straightforward.